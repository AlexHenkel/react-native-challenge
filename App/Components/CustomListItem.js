import React, { PropTypes } from 'react'
import {
  View,
  TextInput,
  TouchableWithoutFeedback
} from 'react-native'
import { connect } from 'react-redux'
import Swipeout from 'react-native-swipeout'
import CheckBox from 'react-native-checkbox'
import Styles from './Styles/CustomListItemStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import Colors from '../Themes/Colors'
import ToDosActions from '../Redux/ToDosRedux'

export class CustomListItem extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func,
    taskInfo: PropTypes.shape({
      index: PropTypes.number,
      title: PropTypes.string,
      completed: PropTypes.bool,
      favorite: PropTypes.bool
    }).isRequired,
    toggleCompletedTask: PropTypes.func.isRequired,
    toggleFavorite: PropTypes.func.isRequired,
    removeTask: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    const { title = '' } = this.props

    this.state = {
      editMode: false,
      editText: title
    }
  }

  _onToggleCheckbox (index) {
    this.props.toggleCompletedTask(index)
  }

  _deleteTask (index) {
    this.props.removeTask(index)
  }

  _addFavorite (index) {
    this.props.toggleFavorite(index)
  }

  _toggleEdit () {
    this.setState({ editMode: !this.state.editMode })
  }

  _handleEditTask = (editText) => this.setState({ editText })

  _handlePressEdit = () => {
    const { editText, editMode } = this.state
    const { index } = this.props.taskInfo
    this.props.editTask(editText, index)
    this.setState({ editMode: !editMode })
  }

  _renderFavorite (favorite) {
    if (favorite) {
      return (
        <Icon style={Styles.favorite} name='star' />
      )
    } else return null
  }

  _renderEditOrCheckbox () {
    const { editMode, editText } = this.state
    const { index = 0, title = '', completed = true, favorite = true } = this.props.taskInfo || {}
    const swipeBtns = [
      {
        text: 'Favorite',
        backgroundColor: 'gold',
        underlayColor: 'gold',
        onPress: () => this._addFavorite(index)
      },
      {
        text: 'Edit',
        backgroundColor: 'blue',
        underlayColor: 'blue',
        onPress: () => this._toggleEdit()
      },
      {
        text: 'Delete',
        backgroundColor: 'red',
        underlayColor: Colors.error,
        onPress: () => this._deleteTask(index)
      }
    ]
    // When edit mode is off, the normal cell is rendered, with the checkbox
    if (!editMode) {
      return (
        <Swipeout
          right={swipeBtns}
          autoClose
          backgroundColor='transparent'>
          <View>
            <CheckBox
              label={title}
              checked={completed}
              onChange={() => this._onToggleCheckbox(index)}
              containerStyle={Styles.listCheckboxContainer}
              checkboxStyle={Styles.listCheckbox}
              labelStyle={Styles.listText}
            />
            {this._renderFavorite(favorite)}
          </View>
        </Swipeout>
      )
    } else {
      // Edit mode is on, so render input
      return (
        <View>
          <View style={Styles.editInput}>
            <TextInput
              style={Styles.editInputText}
              value={editText}
              keyboardType='default'
              returnKeyType='go'
              autoCapitalize='none'
              onChangeText={this._handleEditTask}
              underlineColorAndroid='transparent'
              onSubmitEditing={this._handlePressEdit}
              placeholder='Type to edit task' />
          </View>
          {/* Submit edit button */}
          <TouchableWithoutFeedback onPress={this._handlePressEdit}>
            <Icon style={[Styles.editButtons, Styles.editSubmit]} name='check-circle' />
          </TouchableWithoutFeedback>
          {/* Cancel edit button */}
          <TouchableWithoutFeedback onPress={() => this._toggleEdit()}>
            <Icon style={[Styles.editButtons, Styles.editClose]} name='close' />
          </TouchableWithoutFeedback>
        </View>
      )
    }
  }

  render () {
    return (
      <View style={Styles.listContainer}>
        {this._renderEditOrCheckbox()}
      </View>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCompletedTask: index => dispatch(ToDosActions.toggleCompletedTask(index)),
    removeTask: index => dispatch(ToDosActions.removeTask(index)),
    toggleFavorite: index => dispatch(ToDosActions.toggleFavorite(index)),
    editTask: (task, index) => dispatch(ToDosActions.editTask(task, index))
  }
}

export default connect(null, mapDispatchToProps)(CustomListItem)
