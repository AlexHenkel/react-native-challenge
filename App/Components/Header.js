import React from 'react'
import {
  View,
  Text
} from 'react-native'
import PillButton from '../Components/PillButton'
import Styles from './Styles/HeaderStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'

export default class Header extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      selectedFilter: 'All'
    }
  }

  _onPressFilterAll = () => this.setState({ selectedFilter: 'All' })
  _onPressFilterCompleted = () => this.setState({ selectedFilter: 'Completed' })
  _onPressFilterPending = () => this.setState({ selectedFilter: 'Pending' })

  render () {
    const { selectedFilter } = this.state
    console.log(selectedFilter)
    return (
      <View style={Styles.navBar}>
        <Text style={[Styles.title, Styles.centered]}>myTODOS</Text>
        <Icon style={Styles.add} name='add' />
        <View style={Styles.filters}>
          <PillButton
            text={'All'}
            onPress={this._onPressFilterAll}
            isActive={selectedFilter === 'All'} />
          <PillButton
            text={'Completed'}
            onPress={this._onPressFilterCompleted}
            isActive={selectedFilter === 'Completed'} />
          <PillButton
            text={'Pending'}
            onPress={this._onPressFilterPending}
            isActive={selectedFilter === 'Pending'} />
        </View>
      </View>
    )
  }
}
