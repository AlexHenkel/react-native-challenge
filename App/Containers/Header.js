import React, { PropTypes } from 'react'
import {
  View,
  Text,
  TouchableWithoutFeedback,
  TextInput
} from 'react-native'
import { connect } from 'react-redux'
import { filter } from 'ramda'
import PillButton from '../Components/PillButton'
import Styles from './Styles/HeaderStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import ToDosActions from '../Redux/ToDosRedux'

export class Header extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func,
    addTask: PropTypes.func.isRequired,
    changeFilter: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    this.state = {
      selectedFilter: 'All',
      addMode: false,
      newTask: '',
      pendingLength: 0
    }
  }

  componentWillReceiveProps (nextProps) {
    // Get data from Redux
    const { tasks } = nextProps
    // Get pending tasks length
    const pendingLength = tasks ? filter(n => !n.completed, tasks).length : 0

    // Update dataSource with completely processed data
    this.setState({ pendingLength })
  }

  _onToggleAdd = () => this.setState({ addMode: !this.state.addMode, newTask: '' })

  _handleChangeTask = (newTask) => this.setState({ newTask })
  _handlePressAdd = () => {
    const { newTask, addMode } = this.state
    this.props.addTask(newTask)
    this.setState({ addMode: !addMode, newTask: '' })
  }

  _onPressFilterAll = () => {
    this.props.changeFilter('All')
    this.setState({ selectedFilter: 'All' })
  }
  _onPressFilterCompleted = () => {
    this.props.changeFilter('Completed')
    this.setState({ selectedFilter: 'Completed' })
  }
  _onPressFilterPending = () => {
    this.props.changeFilter('Pending')
    this.setState({ selectedFilter: 'Pending' })
  }

  _renderFiltersOrAdd () {
    const { addMode, selectedFilter, newTask, pendingLength } = this.state
    if (addMode) {
      return (
        <View>
          <View style={Styles.addInput}>
            <TextInput
              style={Styles.addInputText}
              value={newTask}
              keyboardType='default'
              returnKeyType='go'
              autoCapitalize='none'
              onChangeText={this._handleChangeTask}
              underlineColorAndroid='transparent'
              onSubmitEditing={this._handlePressAdd}
              placeholder='Type to add new tasks' />
          </View>
          <TouchableWithoutFeedback onPress={this._handlePressAdd}>
            <Icon style={Styles.addSubmit} name='check-circle' />
          </TouchableWithoutFeedback>
        </View>
      )
    } else {
      return (
        <View style={Styles.filters}>
          <PillButton
            text={'All'}
            onPress={this._onPressFilterAll}
            isActive={selectedFilter === 'All'} />
          <PillButton
            text={'Completed'}
            onPress={this._onPressFilterCompleted}
            isActive={selectedFilter === 'Completed'} />
          <PillButton
            text={`Pending - ${pendingLength}`}
            onPress={this._onPressFilterPending}
            isActive={selectedFilter === 'Pending'} />
        </View>
      )
    }
  }

  render () {
    const { addMode } = this.state
    const addIcon = !addMode ? 'add' : 'close'
    return (
      <View style={Styles.navBar}>
        <Text style={[Styles.title, Styles.centered]}>myTODOS</Text>
        <TouchableWithoutFeedback onPress={this._onToggleAdd}>
          <Icon style={Styles.add} name={addIcon} />
        </TouchableWithoutFeedback>

        {this._renderFiltersOrAdd()}
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    tasks: state.todos.tasks
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addTask: task => dispatch(ToDosActions.addTask(task)),
    changeFilter: filterBy => dispatch(ToDosActions.changeFilter(filterBy))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)
