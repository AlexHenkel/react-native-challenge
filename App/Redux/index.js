import { combineReducers } from 'redux'
import configureStore from './CreateStore'

export default () => {
    /* ------------- Assemble The Reducers ------------- */
  const rootReducer = combineReducers({
    startup: require('./StartupRedux').reducer
  })

  return configureStore(rootReducer)
}
