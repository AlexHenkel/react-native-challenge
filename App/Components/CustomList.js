import React, { PropTypes } from 'react'
import {
  ScrollView,
  RefreshControl,
  ListView
} from 'react-native'
import { connect } from 'react-redux'
import CustomListItem from '../Components/CustomListItem'
import { filter, sort, toLower } from 'ramda'
import Styles from './Styles/CustomListStyles'
import ToDosActions from '../Redux/ToDosRedux'

class CustomList extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func,
    tasks: PropTypes.array,
    filterBy: PropTypes.string,
    toggleCompletedTask: PropTypes.func.isRequired,
    removeTask: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    // This could be faster, but checkbox crashes
    const rowHasChanged = (r1, r2) => r1 !== r2
    const ds = new ListView.DataSource({rowHasChanged})

    this.state = {
      dataSource: ds.cloneWithRows([])
    }
  }

  componentWillReceiveProps (nextProps) {
    // Get data from Redux
    const { tasks, filterBy } = nextProps
    // Get dataSource for ListView
    const { dataSource } = this.state
    // Sort data in descending order by title
    let filteredTasks = sort((a, b) => toLower(a.title) > toLower(b.title), tasks)
    // Apply filters if necessary
    if (filterBy === 'Completed') filteredTasks = filter(n => n.completed, filteredTasks)
    if (filterBy === 'Pending') filteredTasks = filter(n => !n.completed, filteredTasks)

    // Update dataSource with completely processed data
    this.setState({dataSource: dataSource.cloneWithRows(filteredTasks)})
  }

  _renderRow (rowData) {
    return (
      <CustomListItem taskInfo={rowData} />
    )
  }

  render () {
    const { dataSource } = this.state
    const searching = false
    return (
      <ScrollView style={Styles.container}>
        <ListView
          enableEmptySections
          automaticallyAdjustContentInsets={false}
          dataSource={dataSource}
          renderRow={row => this._renderRow(row)}
          refreshControl={
            <RefreshControl
              refreshing={searching}
            />
          }
        />
      </ScrollView>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    tasks: state.todos.tasks,
    filterBy: state.todos.filterBy
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCompletedTask: index => dispatch(ToDosActions.toggleCompletedTask(index)),
    removeTask: index => dispatch(ToDosActions.removeTask(index)),
    resetTasks: () => dispatch(ToDosActions.resetTasks()),
    toggleFavorite: index => dispatch(ToDosActions.toggleFavorite(index))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CustomList)
