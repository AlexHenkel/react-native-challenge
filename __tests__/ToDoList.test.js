import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import App from '../App/Containers/App'
import { Header } from '../App/Components/Header'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('>>>App --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <App />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>Header --- Snapshot', () => {
  it('render correctly', () => {
    const renderedComponent = renderer.create(
      <Header />
    ).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})

describe('>>>Header --- Shallow Render REACT COMPONENTS', () => {
  let wrapper
  const tasks = []

  beforeEach(() => {
    wrapper = shallow(<Header tasks={tasks} />)
  })

  it('render the dumb component', () => {
    expect(wrapper.length).toEqual(1)
  })
})
