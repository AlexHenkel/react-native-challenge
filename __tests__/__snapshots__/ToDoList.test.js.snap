// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>App --- Snapshot render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomColor": "#e32f66",
        "borderBottomWidth": 3,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#161616",
            "fontFamily": "Avenir-Black",
            "fontSize": 18,
            "marginTop": 5,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      myTODOS
    </Text>
    <Text
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#e32f66",
            "fontSize": 24,
            "position": "absolute",
            "right": 10,
            "top": 24,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      testID={undefined}
    >
      
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "rgba(179, 233, 199, 0.1)",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#e32f66",
            "borderColor": "rgba(0, 0, 0, 0.0)",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "overflow": "hidden",
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#FFF",
              },
            ]
          }
        >
          All
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.0)",
            "borderColor": "#DDD",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Completed
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.0)",
            "borderColor": "#DDD",
            "borderRadius": 20,
            "borderWidth": 2,
            "height": 36,
            "marginBottom": 10,
            "marginTop": 10,
            "minWidth": 60,
            "opacity": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#DDD",
                "fontFamily": "Avenir-Black",
                "fontSize": 17,
                "paddingBottom": 5,
                "paddingTop": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Pending - 0
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#D0E0DD",
      }
    }
  >
    <View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e32f66",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginTop": 20,
              "textAlign": "center",
            }
          }
        >
          No tasks yet :(
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e32f66",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginTop": 20,
              "textAlign": "center",
            }
          }
        >
          Let's add something fun
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`>>>CustomList --- Snapshot render correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#D0E0DD",
    }
  }
>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#e32f66",
            "fontSize": 34,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        No tasks yet :(
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#e32f66",
            "fontSize": 34,
            "fontWeight": "bold",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Let's add something fun
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`>>>Header --- Snapshot render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottomColor": "#e32f66",
      "borderBottomWidth": 3,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#161616",
          "fontFamily": "Avenir-Black",
          "fontSize": 18,
          "marginTop": 5,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    myTODOS
  </Text>
  <Text
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#e32f66",
          "fontSize": 24,
          "position": "absolute",
          "right": 10,
          "top": 24,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    testID={undefined}
  >
    
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(179, 233, 199, 0.1)",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e32f66",
          "borderColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "overflow": "hidden",
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderColor": "#DDD",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        Completed
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderColor": "#DDD",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 36,
          "marginBottom": 10,
          "marginTop": 10,
          "minWidth": 60,
          "opacity": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DDD",
              "fontFamily": "Avenir-Black",
              "fontSize": 17,
              "paddingBottom": 5,
              "paddingTop": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        Pending - 0
      </Text>
    </View>
  </View>
</View>
`;
